modelforge/potential/processing.py

```diff
@@ -334,3 +421,92 @@ def forward(self, data: Dict[str, torch.Tensor]) -> torch.Tensor:
+        """
+        Computes the long-range electrostatic energy for a molecular system
+        based on predicted partial charges and pairwise distances between atoms.
+
+        The implementation follows the methodology described in the PhysNet
+        paper, using a cutoff function to handle long-range interactions.
+
+        Parameters
+        ----------
+        strategy : str
+            The strategy to be used for computing the long-range electrostatic
+            energy.
+        cutoff : unit.Quantity
+            The cutoff distance beyond which the interactions are not
+            considered.
+
+        Attributes
+        ----------
+        strategy : str
+            The strategy for computing long-range interactions.
+        cutoff_function : nn.Module
+            The cutoff function applied to the pairwise distances.
+        """
+        super().__init__()
+        from .utils import CosineAttenuationFunction
+
+        self.strategy = strategy
+        self.cutoff_function = CosineAttenuationFunction(cutoff)
+
+    def forward(self, data: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor]:
+        """
+        Forward pass to compute the long-range electrostatic energy.
+
+        This function calculates the long-range electrostatic energy by considering
+        pairwise Coulomb interactions between atoms, applying a cutoff function to
+        handle long-range interactions.
+
+        Parameters
+        ----------
+        data : Dict[str, torch.Tensor]
+            Input data containing the following keys:
+            - 'per_atom_charge': Tensor of shape (N,) with partial charges for each atom.
+            - 'atomic_subsystem_indices': Tensor indicating the molecule each atom belongs to.
+            - 'pairwise_properties': Object containing pairwise distances and indices.
+
+        Returns
+        -------
+        Dict[str, torch.Tensor]
+            The input data dictionary with an additional key 'long_range_electrostatic_energy'
+            containing the computed long-range electrostatic energy.
+        """
+        per_atom_charge = data["per_atom_charge"]
+        mol_indices = data["atomic_subsystem_indices"]
+        pairwise_properties = data["pairwise_properties"]
+        idx_i, idx_j = pairwise_properties.pair_indices
+        pairwise_distances = pairwise_properties.d_ij
+
+        # Initialize the long-range electrostatic energy
+        long_range_energy = torch.zeros_like(per_atom_charge)
+
+        # Apply the cutoff function to pairwise distances
+        phi_2r = self.cutoff_function(2 * pairwise_distances)
+        chi_r = phi_2r * (1 / torch.sqrt(pairwise_distances**2 + 1)) + (
+            1 - phi_2r
+        ) * (1 / pairwise_distances)
+
+        # Compute the Coulomb interaction term
+        coulomb_interactions = (
+            (per_atom_charge[idx_i] * per_atom_charge[idx_j])
+            * chi_r
+            / pairwise_distances
```

Comment:
I don't think this equation is correct.  chi already takes into account the 1/r.  Basically, there is an extra  1/r (i.e., 1/pairwise_distances).  This is also missing coulombs constant which is 1.44 in eV*nm, so, this should be 138.96 in kj/mol nm (with e^2 in there).

Reply:
great catch!